src/Core/Checkout/Customer/Validation/Constraint/CustomerVatIdentification.php

```diff
@@ -25,27 +26,43 @@ class CustomerVatIdentification extends Constraint
     /**
      * @param ?array{countryId: string, shouldCheck?: bool} $options
      *
-     * @deprecated tag:v6.8.0 - Parameter $options will be required and natively typed as array
+     * @deprecated tag:v6.8.0 - reason:new-optional-parameter - $options parameter will be removed, use $countryId and $shouldCheck instead
+     * @deprecated tag:v6.8.0 - reason:new-optional-parameter - $countryId parameter will be required and natively typed as string
      *
      * @internal
      */
-    public function __construct($options = null)
+    #[HasNamedArguments]
+    public function __construct(?array $options = null, ?string $countryId = null, ?bool $shouldCheck = null)
     {
-        if ($options === null) {
-            Feature::triggerDeprecationOrThrow('v6.8.0.0', 'The parameter $options will be required and natively typed as array');
+        if ($options !== null || $countryId === null) {
+            Feature::triggerDeprecationOrThrow(
+                'v6.8.0.0',
+                Feature::deprecatedMethodMessage(self::class, __METHOD__, 'v6.8.0.0', 'Use $countryId argument instead of providing it in $options array')
+            );
         }
 
-        $options ??= [];
+        if ($options === null || Feature::isActive('v6.8.0.0')) {
+            if ($countryId === null) {
+                throw CustomerException::missingOption('countryId', self::class);
+            }
 
-        if (!\is_string($options['countryId'] ?? null)) {
-            throw CustomerException::missingOption('countryId', self::class);
-        }
+            parent::__construct();
 
-        if (isset($options['shouldCheck']) && !\is_bool($options['shouldCheck'])) {
-            throw CustomerException::invalidOption('shouldCheck', 'bool', self::class);
-        }
+            $this->countryId = $countryId;
+            $this->shouldCheck = $shouldCheck ?? false;
```

Comment:
can't we just set the default value to false? :thinking:

Reply:
next major it will be

public function __construct(protected ?string $countryId = null, protected bool $shouldCheck = false)

Reply:
but the deprecation method says nothing about the change from a nullable `shouldCheck` to only allowing boolean value with default false :thinking: